extend-string.js

```diff
@@ -0,0 +1,126 @@
+    if(characterCode >= 65 && characterCode <= 90){ //check if character code is in uppercase using ascii code
+      characterCode += 32 // add 32 to the character code to get its lowercase equivalent code
+    } 
+    result += String.fromCharCode(characterCode); //appending each of the transformed character to the result variable
+  }
+  return result
+};
+
+/* Converts first letter of a string to uppercase */
+String.prototype.ucFirst = function() {
+  if(/[A-Z]/.test(this)) return 'First character already capitalized'
+  return this.charAt(0).toUpper() + this.slice(1) //converts first letter to upper and joins it to the rest of the string characters
+}
+
+/* Checks if a string is a question */
+String.prototype.isQuestion = function() {
+  let regEx = /\w+\?$/g; // pattern to search for strings ending with ?
+  return regEx.test(this)? true: false
+}
+
+/* Pulls out all the words in a string */
+String.prototype.words = function() {
+  let regEx = /[\w-]+/g; // pattern to match a word(s)
+  return this.match(regEx);
+}
+
+/* Counts all the words in a string */
+String.prototype.wordCount = function() {
+  return this.words().length // gets the number of words in string
+}
+
+
+/*Gets currency representation of a string*/
+String.prototype.toCurrency = function() {
+  let regEx = /\d(?=\d{3}\.)/g; //pattern to match digit immediately followed by 3 digits
+  let regSeperator = '$&,';
+  return this.replace(regEx, regSeperator) //replace matched result with specified seperator
+}
+
+
+/*Gets integer from currency*/
+String.prototype.fromCurrency = function() {
+  let regEx = /,/g; //pattern to match the character ","
+  let replacer = '';
+  return this.replace(regEx, replacer) //replace the matched result with an empty string
+}
+
+
+/* Converts each letter of a string to the inverse of its case */
+String.prototype.inverseCase = function() {
+  var result = ''; //empty string to store result
+  var i = 0;
+  while(i < this.length){
+    var char = this.charAt(i); //string character of the current iteration
+    if(/[A-Z]/.test(char)){
+      char = char.toLower(); //converts to lowercase
+    } else {
+      char = char.toUpper()  //converts to uppercase
+    }
+    i++
+    result += char //append each transformed character to result variable
+  }
+  return result
+}
+
+
+/* Converts a string to alternating cases */
+String.prototype.alternatingCase = function() {
+  var chars = this.toLower().split(''); //split the string in lowercase
+  for(i = 0; i < chars.length; i++){
+    if(i % 2 !== 0) {
+      chars[i] = chars[i].toUpper(); //convert current letter to uppercase
+    } else {
+      chars[i] = chars[i].toLower()  //convert current letter to lowercase
+    }
+  }
+  return chars.join('') 
+}
+
+
+/* Converts numbers to its words equivalent */
+String.prototype.numberWords = function() {
+  let result = [];
+  let regEx = /\d/g; //pattern to match all digits
```

Comment:
what happens in a case of 'wedr5rdd'? I will suggest that you consider if you want only numbers such as 1234 or you want a characters within.